tests/ArmTemplates.Tests/Moqs/ApiClients/MockPolicyFragmentClient.cs

```diff
@@ -0,0 +1,25 @@
+﻿// --------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+// --------------------------------------------------------------------------
+
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients.Abstractions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients.PolicyFragments;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities.DataProcessors;
+using Moq;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Tests.Moqs.ApiClients
+{
+    class MockPolicyFragmentClient
+    {
+        public static async Task<IPolicyFragmentsClient> GetMockedHttpPolicyFragmentClient(string responseFileLocation)
+        {
+            var mockedProcessor = new PolicyFragmentDataProcessor();
```

Comment:
I was thinking to remove mocks for dataProcessors where it makes sense, since dataprocessor is an additional step in processing data, it allows us to catch the error on development stage if something would go wrong. 
Apart from that the tests that uses this mockedHttpClient should go from fetching the results till the generation of the template. 
What do you think?